mlprimitives/candidates/keras_LSTM_CycleGAN.py

```diff
@@ -0,0 +1,276 @@
+        partial_gp_loss_x.__name__ = 'gradient_penalty'
+        self.critic_x_model = Model(
+            inputs=[
+                x, z], outputs=[
+                valid_x, fake_x, validity_interpolated_x])
+        self.critic_x_model.compile(
+            loss=[
+                self.wasserstein_loss,
+                self.wasserstein_loss,
+                partial_gp_loss_x],
+            optimizer=optimizer,
+            loss_weights=[
+                1,
+                1,
+                5])
+
+        fake_z = self.critic_z(z_)
+        valid_z = self.critic_z(z)
+        interpolated_z = RandomWeightedAverage()([z, z_])
+        validity_interpolated_z = self.critic_z(interpolated_z)
+        partial_gp_loss_z = partial(self.gradient_penalty_loss, averaged_samples=interpolated_z)
+        partial_gp_loss_z.__name__ = 'gradient_penalty'
+        self.critic_z_model = Model(
+            inputs=[
+                x, z], outputs=[
+                valid_z, fake_z, validity_interpolated_z])
+        self.critic_z_model.compile(
+            loss=[
+                self.wasserstein_loss,
+                self.wasserstein_loss,
+                partial_gp_loss_z],
+            optimizer=optimizer,
+            loss_weights=[
+                1,
+                1,
+                10])
+
+        self.critic_x.trainable = False
+        self.critic_z.trainable = False
+        self.generator.trainable = True
+        self.encoder.trainable = True
+
+        z_gen = Input(shape=(self.latent_dim, 1))
+        x_gen_ = self.generator(z_gen)
+        x_gen = Input(shape=self.shape)
+        z_gen_ = self.encoder(x_gen)
+        x_gen_rec = self.generator(z_gen_)
+        fake_gen_x = self.critic_x(x_gen_)
+        fake_gen_z = self.critic_z(z_gen_)
+
+        self.generator_model = Model([x_gen, z_gen], [fake_gen_x, fake_gen_z, x_gen_rec])
+        self.generator_model.compile(
+            loss=[
+                self.wasserstein_loss,
+                self.wasserstein_loss,
+                'mse'],
+            optimizer=optimizer,
+            loss_weights=[
+                1,
+                1,
+                50])
+
+    def train(self, X, epochs):
+        X_train = X
+        fake = np.ones((self.batch_size, 1))
+        valid = -np.ones((self.batch_size, 1))
+        delta = np.ones((self.batch_size, 1)) * 10
+
+        for epoch in range(epochs):
+            for _ in range(self.n_critic):
+                idx = np.random.randint(0, X_train.shape[0], self.batch_size)
+                x = X_train[idx]
+                z = np.random.normal(size=(self.batch_size, self.latent_dim, 1))
+
+                cx_loss = self.critic_x_model.train_on_batch([x, z], [valid, fake, delta])
+                cz_loss = self.critic_z_model.train_on_batch([x, z], [valid, fake, delta])
+
+            g_loss = self.generator_model.train_on_batch([x, z], [valid, valid, x])
+
+            if epoch % 100 == 0:
+                print(
```

Comment:
Please use `str.format` instead of `+` concatenation. Also curate a bit the formatting. This could be possibly be written in two lines:

```python
print('Epoch: {} [Dx loss:{}] [...'.format(
    epoch, cx_loss...))
```